ex1/buttonActions.js

```diff
@@ -1,13 +1,38 @@
 
 var indexOfColor = 1;
+var colorArray = ['blue', 'red','orange', 'yellow'];
+//A function that changes the color of button on every click
+//If we got to the last color then an image will appear.
 function changeColor() {
-  var colorArray = ['#FF6633', '#CC9999','#4D80CC'];
-    //store
-    sessionStorage.setItem("color", colorArray[indexOfColor]);
-    // Retrieve
-    document.getElementById("demo").style.backgroundColor = sessionStorage.getItem("color");
-
+    setTheColor(indexOfColor);
+    var noaImage = document.getElementById("noaIMG");
     if(++indexOfColor == colorArray.length){
       indexOfColor = 0;
+      noaImage.style.display = "block";
+    } else {
+       noaImage.style.display = "none";
     }
 }
+
+//A function to start the array colors from the start
+function restartArray() {
+  indexOfColor = 0;
+  setTheColor(indexOfColor);
+  document.getElementById("noaIMG").style.display = "none";
+  indexOfColor++;
+
+}
+
+//A function which sets the first color of the button on start
+document.addEventListener("DOMContentLoaded", function(event) {
+  setTheColor(indexOfColor-1);
+});
+
+//A function we call from other functions in
+// order to actually set the color of button
+function setTheColor(indexForArray){
+  //store
+  window.localStorage.setItem("color", colorArray[indexForArray]);
```

Comment:
put colorArray[indexForArray] into a var and use it here and in the next line

Reply:
which next line?

Reply:
line 37-
you use ````window.localStorage.getItem("color")```` and instead you can use ````colorArray[indexForArray] ```` which should be inside a variable as you need to use twice in this function